parsec/core/cli/run.py

```diff
@@ -55,3 +63,173 @@ def run_mountpoint(config, device, mountpoint, timestamp, **kwargs):
         config = config.evolve(mountpoint_base_dir=Path(mountpoint))
     with cli_exception_handler(config.debug):
         trio_run(_run_mountpoint, config, device, timestamp)
+
+
+async def _import_file(workspace_fs: WorkspaceFS, src: FsPath, dest: FsPath):
+    try:
+        await workspace_fs.touch(dest)
+    except FileExistsError:
+        pass
+    with open(src, "rb") as fd:
+        read_size = 0
+        while True:
+            chunk = fd.read(DEFAULT_BLOCK_SIZE)
+            if not chunk:
+                break
+            await workspace_fs.write_bytes(dest, chunk, read_size)
+            read_size += len(chunk)
+
+
+def _local_access_digests(workspace_fs: WorkspaceFS, src: AnyPath):
+    digests = []
+    with open(src, "rb") as fd:
+        read_size = 0
+        while True:
+            chunk = fd.read(DEFAULT_BLOCK_SIZE)
+            if not chunk:
+                break
+            digests.append(HashDigest.from_data(chunk))
+            read_size += len(chunk)
+    return digests
+
+
+async def _update_file(
+    workspace_fs: WorkspaceFS, entry_id: EntryID, path: AnyPath, absolute_path: AnyPath
+):
+    remote_file_manifest = await workspace_fs.remote_loader.load_manifest(entry_id)
+    local_access_digests = _local_access_digests(workspace_fs, path)
+    remote_access_digests = [access.digest for access in remote_file_manifest.blocks]
+    if local_access_digests != remote_access_digests:
+        await _import_file(workspace_fs, path, FsPath(str(absolute_path)))
+        print("update %s" % absolute_path)
+        await workspace_fs.sync_by_id(entry_id, remote_changed=False, recursive=False)
```

Comment:
Here, writing could be optimized to only write the blocks with different digests. This would help with the sync as only the necessary blocks would be uploaded.